code/espurna/thermostat.cpp

```
@@ -488,36 +489,75 @@ void resetBurnCounters() {
 #define mqtt_height 16
+#if THERMOSTAT_DISPLAY_ST7735_SUPPORT
+const char mqtt_bits[] PROGMEM = {
+  0x00, 0x00, 0x00, 0x10, 0x00, 0x18, 0x00, 0x1C, 0x57, 0xFE, 0x57, 0xFE,
+  0x00, 0x1C, 0x08, 0x18, 0x18, 0x10, 0x38, 0x00, 0x7F, 0xEA, 0x7F, 0xEA,
+  0x38, 0x00, 0x18, 0x00, 0x08, 0x00, 0x00, 0x00, };
+#else
 const char mqtt_bits[] PROGMEM = {
   0x00, 0x00, 0x00, 0x08, 0x00, 0x18, 0x00, 0x38, 0xEA, 0x7F, 0xEA, 0x7F,
   0x00, 0x38, 0x10, 0x18, 0x18, 0x08, 0x1C, 0x00, 0xFE, 0x57, 0xFE, 0x57,
   0x1C, 0x00, 0x18, 0x00, 0x10, 0x00, 0x00, 0x00, };
+#endif
 
 #define remote_temp_width 16
 #define remote_temp_height 16
+#if THERMOSTAT_DISPLAY_ST7735_SUPPORT
+const char remote_temp_bits[] PROGMEM = {
+  0x00, 0x00, 0x07, 0x18, 0x08, 0xA4, 0x08, 0xA4, 0x09, 0x98, 0x0A, 0x80,
+  0x0A, 0x80, 0x0B, 0x80, 0x0A, 0x80, 0x0A, 0x80, 0x0B, 0x80, 0x0A, 0x80,
+  0x07, 0x00, 0x07, 0x00, 0x07, 0x00, 0x00, 0x00
+};
+#else
 const char remote_temp_bits[] PROGMEM = {
   0x00, 0x00, 0xE0, 0x18, 0x10, 0x25, 0x10, 0x25, 0x90, 0x19, 0x50, 0x01,
   0x50, 0x01, 0xD0, 0x01, 0x50, 0x01, 0x50, 0x01, 0xD0, 0x01, 0x50, 0x01,
   0xE0, 0x00, 0xE0, 0x00, 0xE0, 0x00, 0x00, 0x00, };
+#endif
 
 #define server_width 16
 #define server_height 16
+#if THERMOSTAT_DISPLAY_ST7735_SUPPORT
+const char server_bits[] PROGMEM = {
+  0x00, 0x00, 0x1F, 0xF8, 0x3F, 0xFC, 0x30, 0x0C, 0x30, 0x0C, 0x30, 0x0C,
+  0x30, 0x0C, 0x30, 0x0C, 0x30, 0x0C, 0x1F, 0xF8, 0x3F, 0xFC, 0x7F, 0xFE,
+  0x78, 0x1E, 0x7F, 0xFE, 0x3F, 0xFC, 0x00, 0x00, };
+#else
 const char server_bits[] PROGMEM = {
   0x00, 0x00, 0xF8, 0x1F, 0xFC, 0x3F, 0x0C, 0x30, 0x0C, 0x30, 0x0C, 0x30,
   0x0C, 0x30, 0x0C, 0x30, 0x0C, 0x30, 0xF8, 0x1F, 0xFC, 0x3F, 0xFE, 0x7F,
   0x1E, 0x78, 0xFE, 0x7F, 0xFC, 0x3F, 0x00, 0x00, };
+#endif
 
 #define LOCAL_TEMP_UPDATE_INTERVAL 60000
 #define LOCAL_HUM_UPDATE_INTERVAL  61000
 
+#if THERMOSTAT_DISPLAY_ST7735_SUPPORT
+/*#define cs   D8
+#define dc   D3
+#define rst  D4*/
+#define cs   D8
+#define dc   D4
+#define rst  -1
```

D# are board-specific definitions, maybe a custom setting in general.h?
`#define DISPLAY_ST7735_{RST,CS,DC} ...` 

Similar, for the ssd init, as it is silently using tx and rx  :/

---

There should me more customization... eg: rotation.. 
There are lot of different similar display that may need some different specifications...
Maybe better in platformio_override?

Will try with general...

---

general header should have things regardless, as reference / defaults. ifndef check allows us to override things from custom.h / platformio_override

---

done as you suggested!